### DISCUSSION TITLE
Matek FC's not working with O3

### DISCUSSION THREAD

**Discussion by dcan999** [iNavFlight/inav#9949] [Support] [answered]
I posted my main issues last night but realized it was in the Configurator section, not sure which section better suits the issue. 

Anyways, I did a lot more testing today and have confirmed the following. 

My Vista systems work (show OSD) on all my FCs (Speedybee and Matek) and all my configs, I get OSD in the goggles. 

My O3 systems ONLY work (show OSD) on my SpeedyBee FC's, on all configs. 

I have fresh installs on 7.1.0 on all of the FC's, and it doesn't matter what configs I swap (diff all and even dump all) to what FC, the OSD always works on the SpeedyBee + O3, and never works on Matek + O3. 

To do the testing I stopped using any pre configured diff files and just made a fresh config with just the two ports in use (O3 and serial RX) that I used on all of the FC's just to keep everything the same and rule out configuration issues. 

The FC's I've tried at 

Matek F405-WMN (three different ones)
Matek H743-Wing V3
Matek F405-Wing V2

SpeedyBee F4 Mini Stack
Speedybee F405-Wing (that I just got today to test my theory)

Two different Vistas (one used daily, one new in box)
Three different O3 (one used last year on 6.1.0, two brand new)

I'm at a loss at this point wondering if it's still something I'm doing or if there could be a problem with the Matek targets (I don't know enough to make that presumption). 

I had my oldest O3 working fine last year on one of the 405-WMN's, I believe it was on 6.0 and 6.1, before I crashed it and put it away for the winter, but when I rebuilt it recently, I didn't even try and fly it before I upgraded to 7.0.0, so I don't know if that could be part of the issue. 

I did try and downgrade one of the 405-WMN to 6.0 using the 6.0 configurator, but the O3 OSD still didn't work, so I'm really at a loss. 

Any insight would be greatly appreciated.

**Comment by DzikuVx** [comment]
O3 works with Matek FCs. I know because I mostly fly Matek and O3 with INAV. You are doing something wrong with configuration, but telling what is impossible because you have not attached any diff of your config

**Reply by dcan999** [reply, marked as answer]
Thanks for the reply. I was double checking what config was on what FC before replying and when moving them around I realized I was using the pre-made PNP connectors for the Speedybee FC's, and regular connectors for the Mateks. I actually made note of that during testing, but I triple checked the three different cables I was using, BUT, I checked continuity by back probing the SH connector going into the O3, and probing the pin the Dupont was plugged into on the FC, which showed continuity, so I presumed all cables were good. 

Today I figured it was slightly possible even if there was continuity at the back of the SH socket, it might not be connecting with the O3 pin when plugged in, so I plugged the cables into a blank female SH socket and checked again. ALL THREE cables had no continuity on either the rx or tx wires, the pin sockets were slightly bent open and not making contact with the O3 pins.
